src/compiler/transformers/esDecorators.ts

```diff
@@ -220,6 +220,7 @@ interface ClassInfo {
     classExtraInitializersName?: Identifier; // used in step 13
     classThis?: Identifier; // `_classThis`, if needed.
     classSuper?: Identifier; // `_classSuper`, if needed.
+    metadataReference?: Identifier;

```

Comment:
If we are transforming a class with decorators, we will always have a `metadataReference`, so it should be assigned in `createClassInfo()`